### ISSUE TITLE
Inquiry Regarding First Stage Training and Test Results

### ISSUE THREAD

**Issue by Emily-29** [yuhuUSTC/DehazeDDPM#5]
Thank you for making your code accessible on GitHub. It has been invaluable for my project on image dehazing. I have a query regarding the initial phase of your model's training process that I hope you can clarify: could you elaborate on how the training is conducted during the first stage?

Furthermore, I attempted to apply the models "NH_net_g_80000.pth" and "NH_I230000_E4600_gen.pth" to image "55.png" from the NH-HAZE dataset. The outcomes did not align with my expectations. I would greatly appreciate it if you could review the attached results and advise if there might be any steps I have overlooked or misimplemented.

![ret](https://github.com/yuhuUSTC/DehazeDDPM/assets/63189342/d7671c8f-26b5-4dbe-b579-c69ad3c45d33)

**Comment by yeziyuan27** [comment]
Hello! I have also encountered this problem. Have you solved this problem?

**Comment by yuhuUSTC** [comment]
The first stage follows the well-known ASM formulation: I(x) = J(x)t(x) + A (1 - t(x)). For the output in https://github.com/yuhuUSTC/DehazeDDPM/blob/5ee18db0646de74741714c83493a8d3f17c1a8c2/model/networkHelper.py#L676,  out_J means the final output, stage1_output represents the output of the first stage, out_T denotes t(x), out_A denotes A, and out_I denotes I(x). 
Therefore, the training loss for the first stage is: loss = L1loss(HR, out_J) + L1loss(HR, stage1_output) + L1loss(LR, out_I).

**Comment by yeziyuan27** [comment]
Thank you very much for your reply!
I would like to know why the result of my validation with your pre-trained model on the 55-hazy images on the NH-haze dataset is a noised image?This question is the same as the second question raised by the host, looking forward to your reply!  Thank you very much!
![屏幕截图 2024-08-16 210107](https://github.com/user-attachments/assets/830654d2-e2f2-42de-a7a7-81079d8e470a)
![屏幕截图 2024-08-16 210210](https://github.com/user-attachments/assets/5c0bd1d9-b023-4587-a276-3a20496e9ad3)
@yuhuUSTC

**Comment by Emily-29** [comment]
> Hello! I have also encountered this problem. Have you solved this problem?

Not yet

**Comment by zhengchaobing** [comment]
> Hello! I have also encountered this problem. Have you solved this problem?

I have also encountered this problem too. Have you solved this problem now?

**Comment by zhengchaobing** [comment]
> Thank you very much for your reply! I would like to know why the result of my validation with your pre-trained model on the 55-hazy images on the NH-haze dataset is a noised image?This question is the same as the second question raised by the host, looking forward to your reply! Thank you very much! ![屏幕截图 2024-08-16 210107](https://private-user-images.githubusercontent.com/169962076/358590369-830654d2-e2f2-42de-a7a7-81079d8e470a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NTA5ODIsIm5iZiI6MTcyODQ1MDY4MiwicGF0aCI6Ii8xNjk5NjIwNzYvMzU4NTkwMzY5LTgzMDY1NGQyLWUyZjItNDJkZS1hN2E3LTgxMDc5ZDhlNDcwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwNTExMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDYwN2NkNGNjMzA4NDU2YjJiNTdjMzY2YmNlZDJlYjU1NjkxM2Q0NWU4YWQzODUxNmU4NTJjNGU2NmNkOTMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.y4lQ3aRahsNNxp9qrZ44gpAgA3VoqLz95dK00xRXpwo) ![屏幕截图 2024-08-16 210210](https://private-user-images.githubusercontent.com/169962076/358590593-5c0bd1d9-b023-4587-a276-3a20496e9ad3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NTA5ODIsIm5iZiI6MTcyODQ1MDY4MiwicGF0aCI6Ii8xNjk5NjIwNzYvMzU4NTkwNTkzLTVjMGJkMWQ5LWIwMjMtNDU4Ny1hMjc2LTNhMjA0OTZlOWFkMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwNTExMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZDY1MGFjNDM5YmJmN2I2NWY5OWQ5M2Y0MjA2YmJlMGEwOWQ1YTc4YWFiNmJhZWRhNDZiZWEyMzA4NDYwYzc5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EWzN7tx4XpyoMjlQcLpDDlzduXYsvrCSsAgG5-DJSvY) @yuhuUSTC

I have also encountered this problem too. Have you solved this problem now?

**Comment by ZhuLiangyu123** [comment]
> 感谢您在 GitHub 上提供可访问的代码。它对我的图像去雾项目非常宝贵。关于您的模型训练过程的初始阶段，我有一个疑问，希望您能澄清一下：您能详细说明一下第一阶段的训练是如何进行的吗？
> 
> 此外，我尝试将模型 “NH_net_g_80000.pth” 和 “NH_I230000_E4600_gen.pth” 应用于 NH-HAZE 数据集中的图像 “55.png”。结果与我的预期不一致。如果您能查看随附的结果并建议我是否遗漏或错误实施任何步骤，我将不胜感激。
> 
> ![ret](https://github.com/yuhuUSTC/DehazeDDPM/assets/63189342/d7671c8f-26b5-4dbe-b579-c69ad3c45d33)

您好，我遇到了和您一样的问题，请问您解决了没？

**Comment by ZhuLiangyu123** [comment]
> 你好！我也遇到过这个问题。你解决了这个问题吗？

您好，我遇到了和您一样的问题，请问您解决了没？

**Comment by ZhuLiangyu123** [comment]
> 非常感谢您的回复！我想知道为什么我在 NH-haze 数据集上的 55 个朦胧图像上使用您的预训练模型进行验证的结果是噪声图像？这个问题和主持人提出的第二个问题一样，期待你的回答！谢谢！ ![屏幕截图 2024-08-16 210107](https://github.com/user-attachments/assets/830654d2-e2f2-42de-a7a7-81079d8e470a) ![屏幕截图 2024-08-16 210210](https://github.com/user-attachments/assets/5c0bd1d9-b023-4587-a276-3a20496e9ad3)

您好，我遇到了和您一样的问题，请问您解决了没？

**Comment by ZhuLiangyu123** [comment]
> > 非常感谢您的回复！我想知道为什么我在 NH-haze 数据集上的 55 个朦胧图像上使用您的预训练模型进行验证的结果是噪声图像？这个问题和主持人提出的第二个问题一样，期待你的回答！谢谢！![屏幕截图 2024-08-16 210107](https://private-user-images.githubusercontent.com/169962076/358590369-830654d2-e2f2-42de-a7a7-81079d8e470a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NTA5ODIsIm5iZiI6MTcyODQ1MDY4MiwicGF0aCI6Ii8xNjk5NjIwNzYvMzU4NTkwMzY5LTgzMDY1NGQyLWUyZjItNDJkZS1hN2E3LTgxMDc5ZDhlNDcwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwNTExMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDYwN2NkNGNjMzA4NDU2YjJiNTdjMzY2YmNlZDJlYjU1NjkxM2Q0NWU4YWQzODUxNmU4NTJjNGU2NmNkOTMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.y4lQ3aRahsNNxp9qrZ44gpAgA3VoqLz95dK00xRXpwo)  ![屏幕截图 2024-08-16 210210](https://private-user-images.githubusercontent.com/169962076/358590593-5c0bd1d9-b023-4587-a276-3a20496e9ad3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NTA5ODIsIm5iZiI6MTcyODQ1MDY4MiwicGF0aCI6Ii8xNjk5NjIwNzYvMzU4NTkwNTkzLTVjMGJkMWQ5LWIwMjMtNDU4Ny1hMjc2LTNhMjA0OTZlOWFkMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwNTExMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZDY1MGFjNDM5YmJmN2I2NWY5OWQ5M2Y0MjA2YmJlMGEwOWQ1YTc4YWFiNmJhZWRhNDZiZWEyMzA4NDYwYzc5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EWzN7tx4XpyoMjlQcLpDDlzduXYsvrCSsAgG5-DJSvY)
> 
> 我也遇到过这个问题。您现在解决了这个问题吗？

您好，我遇到了和您一样的问题，请问您解决了没？

**Comment by yyyyyyounger** [comment]
https://github.com/yuhuUSTC/DehazeDDPM/blob/5ee18db0646de74741714c83493a8d3f17c1a8c2/model/model.py#L150-L163

I think it's the `pop` in line 161 here that causes the input convolution not to be loaded and remains randomly initialized, resulting in pure noise.
```
net.pop("denoise_fn.downs.0.weight") 
```

I modified it to:
```
net = torch.**load**(gen_path)
# net.pop("denoise_fn.downs.0.weight")
network.load_state_dict(net, strict=False)
```
Then, I get the basic result(200 steps):

<img width="672" height="512" alt="Image" src="https://github.com/user-attachments/assets/5caf4fed-f8d2-4811-80c2-6f7002dd2b3f" />
